mobile_verifier/src/sp_boosted_rewards_bans.rs

```diff
@@ -0,0 +1,654 @@
+            file_upload,
+            concat!(env!("CARGO_PKG_NAME"), "_verified_sp_boosted_rewards_ban"),
+        )
+        .auto_commit(false)
+        .create()
+        .await?;
+
+        let (receiver, ingest_server) = FileInfoPollerConfigBuilder::<
+            ServiceProviderBoostedRewardsBannedRadioIngestReportV1,
+            _,
+            _,
+        >::default()
+        .parser(ProstFileInfoPollerParser)
+        .state(pool.clone())
+        .store(file_store)
+        .lookback(LookbackBehavior::StartAfter(settings.start_after))
+        .prefix(FileType::ServiceProviderBoostedRewardsBannedRadioIngestReport.to_string())
+        .create()
+        .await?;
+
+        let ingestor = Self {
+            pool,
+            authorization_verifier,
+            receiver,
+            verified_sink,
+            seniority_update_sink,
+        };
+
+        Ok(TaskManager::builder()
+            .add_task(verified_sink_server)
+            .add_task(ingest_server)
+            .add_task(ingestor)
+            .build())
+    }
+
+    async fn run(mut self, mut shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("service provider boosted rewards ban ingestor starting");
+
+        loop {
+            tokio::select! {
+                biased;
+                _ = &mut shutdown => break,
+                Some(file) = self.receiver.recv() => {
+                    self.process_file(file).await?;
+                }
+            }
+        }
+        tracing::info!("stopping service provider boosted rewards ban ingestor");
+
+        Ok(())
+    }
+
+    async fn process_file(
+        &self,
+        file_info_stream: FileInfoStream<ServiceProviderBoostedRewardsBannedRadioIngestReportV1>,
+    ) -> anyhow::Result<()> {
+        tracing::info!(file = %file_info_stream.file_info.key, "processing sp boosted rewards ban file");
+        let mut transaction = self.pool.begin().await?;
+        file_info_stream
+            .into_stream(&mut transaction)
+            .await?
+            .map(anyhow::Ok)
+            .try_fold(transaction, |mut tx, ingest| async move {
+                self.process_ingest_report(&mut tx, ingest).await?;
+                Ok(tx)
+            })
+            .await?
+            .commit()
+            .await?;
+
+        self.verified_sink.commit().await?;
+        self.seniority_update_sink.commit().await?;
+
+        Ok(())
+    }
+
+    async fn process_ingest_report(
+        &self,
+        transaction: &mut Transaction<'_, Postgres>,
+        ingest: ServiceProviderBoostedRewardsBannedRadioIngestReportV1,
+    ) -> anyhow::Result<()> {
+        let report = BannedRadioReport::try_from(ingest.clone())?;
+        let is_authorized = self.is_authorized(&report.pubkey).await?;
+
+        if is_authorized {
+            db::update_report(transaction, &report).await?;
+            self.update_seniority(transaction, &report).await?;
+        }
+
+        let status = match is_authorized {
```

Comment:
I do think that an if statement is cleaner here

Reply:
Addressed 15874cb